### DISCUSSION TITLE
Is it safe to periodically remove expired sessions in `session` table?

### DISCUSSION THREAD

**Discussion by joarobles** [apache/airflow#27751] [Q&A]
We have only a few DAGs running for a couple of months and I can see the `session` table is becoming very large (+2M rows) and most rows have the `expiry` column with a `datetime` value less than the current date.

1. Is it safe to periodically remove expired sessions in `session` table?
2. When are this sessions created?
3. Once expired, are they reused?

**Comment by potiuk** [comment]
yep Should be safe. The entries are created by Flask Application Builder when you login. Not sure why you have som many of them  though - likely some automated process that keeps on logging in continuously (which should rather re-use existing connection/session token than re-login). so you might look at that.

You can also go ask in FAB https://flask-appbuilder.readthedocs.io/en/latest/views.html - because this part of Airlfow comes directly from FAB.

**Reply by joarobles** [reply]
Thanks @potiuk for your reply!
I'm trying to keep track on this, sessions are still created every second or so, 
How can I audit which user/s are logging in? The `sessions` table doesn't show that information

**Reply by Taragolis** [reply]
This table only use for store Flask session into database: https://github.com/apache/airflow/pull/21478 and not intend to use for any of audit.

I'm also not sure that we store information about is user login or not but I could be wrong and miss something. 
You could check how many success/failed logins made by specific user: in UI: `Security > User's Statistics` or check `ab_user` table

There is a bit strange that you have session which created every seconds maybe you have use some specific setup or usage of Airflow?

**Reply by averni** [reply]
Hi,

we had similar issues when running webserver health checks against paths different from /health.  
Each run creates a new session after being redirected to the login page.

**Reply by joarobles** [reply]
Thanks @averni and @Taragolis, that was the problem indeed: AWS ALB health checks were checking the root path "/".

As the health check endpoint /health is not protected all requests to that path won't create a session.

**Reply by potiuk** [reply]
Oh. nice one! TIL.
